pilot/pkg/util/factory/hash.go

```diff
@@ -0,0 +1,10 @@
+package factory
+
+import (
+	"github.com/cespare/xxhash/v2"
+	"hash"
+)
+
+func NewHash() hash.Hash {
+	return xxhash.New()
```

Comment:
I like abstracting this out but `factory` is not a great package name as it is pretty vague. maybe `hash.New`?

We could also return a wrapper that is easier to use an has methods like `ToString()` and `ToInt()` to encode it instead of hex.EncodeToString etc

Reply:
This is too simple a wrapper and without customization, is it worthwhile?

IMO we can simply call xxhash.New()

Reply:
both said reasonable, what should i do ?